packages/workbox-background-sync/models/QueueStore.mjs

```diff
@@ -60,55 +58,19 @@ export class QueueStore {
    * value as a StorableRequest instance. If no entry exists, it returns
    * undefined.
    *
-   * @return {Promise<StorableRequest|undefined>}
+   * @return {StorableRequest|undefined}
```

Comment:
Yeah, I was thinking about bringing this up to the group as a style guide issue.

Initially I was all for being explicit about the fact that async functions return promises, but now I'm starting to lean the other way (partially since it plays better with current tools).

The other thing I've been doing is not mixing async and promises. I.e. even if I have a variable that I know is a promise, I `return await` it, since my understanding is it leads to [better stack traces](https://mathiasbynens.be/notes/async-stack-traces).

Anyway, I guess this is a larger discussion...

Reply:
(As an aside, https://eslint.org/docs/rules/no-return-await is not enabled, so there's nothing from the linting perspective stopping up from using `return await...`.)

Reply:
For internal functions it's not an issue but for developer facing docs it needs to be returning a promise regardless of anything internally

Reply:
Agreed for public facing functions, though It would be great if the docs could automatically detect that it's an async function and indicate that accordingly.